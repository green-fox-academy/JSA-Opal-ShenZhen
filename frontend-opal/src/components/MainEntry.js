import React, { useContext } from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';
import { NavigationContext } from 'react-navigation';

import tools from './common/tools';

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: '100%',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center'
  },
  start: {
    fontSize: tools.getPixel(36),
    color: 'rgb(77,119,245)'
  },
  your: {
    fontSize: tools.getPixel(36),
    color: 'rgb(77,119,245)'
  },
  investment: {
    fontSize: tools.getPixel(36),
    color: 'rgb(77,119,245)'
  }
});

function MainEntry() {
  const navigation = useContext(NavigationContext);

  return (
    <TouchableOpacity style={styles.container} onPress={() => navigation.navigate('Portfolio')}>
      <Text style={styles.start}>Start</Text>
      <Text style={styles.your}>Your</Text>
      <Text style={styles.investment}>Investment</Text>
    </TouchableOpacity>
  );
}

export default MainEntry;
