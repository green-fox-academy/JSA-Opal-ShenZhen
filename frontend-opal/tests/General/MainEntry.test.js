import React from 'react';
import renderer from 'react-test-renderer';
import MainEntry from 'components/MainEntry';

describe('<MainEntry />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<MainEntry />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
