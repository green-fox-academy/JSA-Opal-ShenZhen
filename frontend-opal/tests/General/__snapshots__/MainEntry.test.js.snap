// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainEntry /> renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "100%",
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgb(77,119,245)",
        "fontSize": 72,
      }
    }
  >
    Start
  </Text>
  <Text
    style={
      Object {
        "color": "rgb(77,119,245)",
        "fontSize": 72,
      }
    }
  >
    Your
  </Text>
  <Text
    style={
      Object {
        "color": "rgb(77,119,245)",
        "fontSize": 72,
      }
    }
  >
    Investment
  </Text>
</View>
`;
